pallets/investments/src/lib.rs

```diff
@@ -499,10 +501,36 @@ pub mod pallet {
 			Self::do_collect_both(who, investment_id)
 		}
 
-		/// Collect the results of another users orders for the given investment.
-		///
-		/// The type of collection will always be `CollectType::Closing`.
-		#[pallet::weight(0)]
+		/// Collect the results of a users invest orders for the given investment.
+		/// If any amounts are not fulfilled they are directly appended to the next active
+		/// order for this investment.
+		#[pallet::weight(80_000_000)]
+		pub fn collect_investments(
```

Comment:
Why do we need these? It seems like this is just adding complexity, while just `collect()` would suffice?

Reply:
we are solely adding a public api endpoint. The logic is there anyways. The frontend should be able to see, if only investments should be collected and then call this endpoint. If there are both redemptions and investments to be collected it should call the other one. From a user perspective always collecting both is forcing them to pay for something they do not need. 

In cases, where this would duplicate logic or introduce complexity I wouldn't add this, but this is really just the endpoint.

Reply:
Yeah that makes sense. But, shouldn't we just remove `collect()` then, and users can just batch the other two methods if needed? Given that the weight of `collect` is simply `weight(collect_investments) + weight(collect_redemptions)`?

Reply:
Yes, that is a good idea. For the collect_for I think forcing collecting of both is also not necessarily needed. I thought about any advantage one could have by only collecting one part for others. But there isn't actually one, when I think about it now. Wdyt?

Reply:
I can't think of any either, I think just rewriting that to `collect_investments/redemptions_for` would be fine.